rasa/core/training/story_reader/markdown_story_reader.py

```diff
@@ -193,25 +198,58 @@ async def _add_e2e_messages(self, e2e_messages: List[Text], line_num: int) -> No
                 "location. Expected story start."
                 "".format(e2e_messages)
             )
-        e2e_reader = EndToEndReader()
+
         parsed_messages = []
         for m in e2e_messages:
-            message = e2e_reader._parse_item(m)
-            parsed = await self._parse_message(message.text, line_num)
-
-            parsed.parse_data["true_intent"] = message.data["true_intent"]
-            parsed.parse_data["true_entities"] = message.data.get("entities") or []
+            message = self.parse_e2e_message(m)
+            parsed = self._parse_message(message.text, line_num)
             parsed_messages.append(parsed)
         self.current_step_builder.add_user_messages(parsed_messages)
 
-    async def _parse_message(self, message: Text, line_num: int) -> UserUttered:
+    @staticmethod
+    def parse_e2e_message(line: Text) -> "Message":
+        f"""Parses an md list item line based on the current section type.
+
+        Matches expressions of the form `<intent>:<example>. For the
+        syntax of <example> see the Rasa docs on NLU training data:
+        {DOCS_BASE_URL}/nlu/training-data-format/#markdown-format"""
+
+        # Match three groups:
+        # 1) Potential "form" annotation
+        # 2) The correct intent
+        # 3) Optional entities
+        # 4) The message text
+        form_group = fr"({FORM_PREFIX}\s*)*"
+        item_regex = re.compile(r"\s*" + form_group + r"([^{}]+?)({.*})*:\s*(.*)")
+        match = re.match(item_regex, line)
+
+        if not match:
+            raise ValueError(
+                "Encountered invalid end-to-end format for message "
+                "`{}`. Please visit the documentation page on "
+                "end-to-end testing at {}/user-guide/testing-your-assistant/"
+                "#end-to-end-testing/".format(line, DOCS_BASE_URL)
+            )
```

Comment:
might make more sense in this case just to link to the stories training data format?

Reply:
well, the new page won't contain the markdown info anyways - will link to the legacy docs instead